### ISSUE TITLE
Touch events in CefSharp.WinForms component hosted in WindowsFormsHost of WPF application.

### ISSUE THREAD

**Issue by dmitryshunkov** [cefsharp/CefSharp#2338]
- **What version of the product are you using?**
`Install-Package CefSharp.WinForms -Version 63.0.3`

- **What architecture x86 or x64?**
x64
- **On what operating system?**
Windows 10

- **Are you using `WinForms`, `WPF` or `OffScreen`?**
I'm using WPF application, but with WindowsFormsHost as this [article](https://github.com/cefsharp/CefSharp/wiki/General-Usage) suggests to get touch support in WPF application.
- **What steps will reproduce the problem?**
I forked CefSharp.MinimalExample and pushed reproducing commit: [branch](https://github.com/dmitryshunkov/CefSharp.MinimalExample/tree/wpf-host-touch-bug).

Steps:
1. Start CefSharp.MinimalExample.Wpf application (google.com will open).
2. **Touch** google search input field (it will get focus), type something
3. Click or touch on any control out of WindowsFormsHost (any WPF control, for example address bar at top of window) to browser lost focus.
4. Again **touch** google search input field, try to type something.
5. Browser and input box didn't get focus.

- **What is the expected output? What do you see instead?**
Touch event should focus browser control and input box.

- **Please provide any additional information below.**
To emulate touch events on windows I used Microsoft Windows Simulator (c:\Program Files (x86)\Common Files\microsoft shared\Windows Simulator\15.0\). It is a part of Windows SDK. If I'm correct, Universal Windows Platform development component should be installed to get the simulator in this folder.

    - Does the cef log provide any relevant information? (By default there should be a debug.log file in your bin directory)
No
    - Any other background information that's relevant? Are you doing something out of the ordinary? 3rd party controls?
No, just WPF minimal example modified with WindowsFormsHost

- **Does this problem also occur in the `CEF` Sample Application from http://opensource.spotify.com/cefbuilds/index.html?**
Unfortunately, it seems that it reproducible on CEF client. And also I found a very old [issue](https://bitbucket.org/chromiumembedded/cef/issues/1797/windows-touch-does-not-return-focus-to-the) with minor priority. But I'm curious if it can be workarounded with some code (may be proper passing messages in handler). If not, I will create issue there.
    - Version: `cef_binary_3.3325.1751.ge5b78a5_windows64_client`

**Comment by amaitland** [comment]
You may have better luck integrating `CEF` into your main applications message loop, see https://github.com/cefsharp/CefSharp/wiki/General-Usage#multithreadedmessageloop

The `WinForms` version also uses a `IFocusHandler` implementation to bridge the gap between the message loops, it's implementation is not perfect, see https://github.com/cefsharp/CefSharp/blob/cefsharp/63/CefSharp.WinForms/Internals/DefaultFocusHandler.cs for the source, you might have to debug it yourself. If you do integrate `CEF` into your apps message loop then you can just set `browser.FocusHandler = null`, you shouldn't need a handler in that scenario.

`CEF` does focus as you've already noted (also has activation issues), so could also be a `CEF` issue.

**Comment by dmitryshunkov** [comment]
Thank you for your answer. Actually, before creating the issue I had already tried to fix it with your advices from other threads. And I have already tried `MultiThreadedMessageLoop = false` with some modification of https://github.com/cefsharp/CefSharp/blob/cefsharp/63/CefSharp.Example/Handlers/BrowserProcessHandler.cs.
I added [commit](https://github.com/dmitryshunkov/CefSharp.MinimalExample/commit/dc1c8f388f02e6224ecceae7f71cdace0891f287) to reproduce issue with integrated message loop on CefSharp.MinimalExample. Also, I tried to force Focus() browser control on focused node changed with RenderProcessMessageHandler, but with no luck. Only working solution was to call Focus() periodically with timer, but it fails for new browser windows, as it focus main window.
